pkg/integrationtest/run.go

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright (C) 2011-2021 Red Hat, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package integrationtest
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/commonjava/indy-tests/pkg/common"
+)
+
+const TARGET_DIR = "target"
+
+/*
+ * When we start the integration test, it will (in order):
+ *
+ * a. Load the corresponding dataset
+ * b. Retrieve the metadata files in da.json to make sure they can be downloaded successfully
+ * c. Create a temp group which is same to PNC temp build group (with a hosted repo, namely A)
+ * d. Download the files in tracking "downloads" section from this temp group
+ * e. Download the files in tracking "uploads" section, and rename all the files (jar, pom, and so on)
+ *    with a new version suffix and upload them again to the hosted repo A
+ * f. Retrieve the metadata files that will be updated by below promotion (we need to foresee those metadata files)
+ * g. Promote the files in hosted repo A to hosted repo pnc-builds.
```

Comment:
Will we promote the test ones into pnc-builds ? or use a dedicated repo ?

Reply:
If we use pnc-builds, we can use rollback promotion to clear the test files. But I prefer a dedicated one if the effort is not bigger. :)

Reply:
the effort is not minor... if the rollback works well, let's try this first.

Reply:
I agree to use a dedicated one which will not pollute the original pnc-builds. But not sure if the dedicated one can help us to test the metadata generation for both correctness and performance.

Reply:
mimic the repo structure is one place that might go wrong overtime. so let's try the rollback way. Although rollback is not used very often, it will be nice to test it by our integration test.